wallet/src/actors/crypto/mod.rs

```diff
@@ -73,6 +78,24 @@ impl Crypto {
             }
         }
     }
+
+    /// Generate a Session ID for an unlocked wallet
+    pub fn gen_session_id(&self, key: &wallet::Key) -> String {
+        match self.id_hash_function {
+            HashFunction::Sha256 => {
+                let rand_bytes: [u8; 32] = self.rng.borrow_mut().gen();
+                let password =
+                    [key.secret.as_ref(), key.salt.as_ref(), rand_bytes.as_ref()].concat();
+                let id_bytes = pbkdf2_sha256(
+                    password.as_ref(),
+                    self.master_key_salt.as_ref(),
+                    self.id_hash_iterations,
+                );
+
+                hex::encode(id_bytes)
+            }
+        }
+    }
```

Comment:
@mariocao @aesedepece @girazoki 📣️ Please, can you review this function? This is the function used to generate session ids for unlocked wallets. Here `key` is the [Key](https://github.com/witnet/witnet-rust/pull/723/files#diff-a4ea0d2d0f3ed43e241630042951988aR93) generated from the password used to encrypt/decrypt the wallet in the storage.